### ISSUE TITLE
mingw-w64 SHA256 mismatch

### ISSUE THREAD

**Issue by adfoster-r7** [Homebrew/homebrew-core#171199]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
vagrant@osx-10_7 ~ % brew install mingw-w64
.... etc ....
Installing mingw-w64 leads to the following issue:


Error: mingw-w64: SHA256 mismatch
Expected: 968bfcb58f75d889470f2f815787f6aa254fb43f1e5516e04f577dad22259905
  Actual: 7f79e6bf59d58562990baaed5a69f83df57e6d6255a3ddf5c779320e58dd5f30
    File: /Users/vagrant/Library/Caches/Homebrew/downloads/6c68aa4aef135895a658b1ab6d97313af8eadb8add341ff7a61b747317d00608--gcc.git-9970b576b7e4ae337af1268395ff221348c4b34a.patch
To retry an incomplete download, remove the file above.

```

Potentially caused by:

https://github.com/Homebrew/homebrew-core/pull/169892

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Install mingw-w64, for the compiler toolchain

### What happened (include all command output)?

```
vagrant@osx-10_7 ~ % brew install mingw-w64
==> Downloading https://formulae.brew.sh/api/formula.jws.json
################################################################################################################################################################## 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
################################################################################################################################################################## 100.0%
Warning: You are using macOS 11.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.

==> Fetching dependencies for mingw-w64: isl, mpfr, libmpc, lz4, cmake, zstd and texinfo
==> Fetching isl
==> Downloading https://ghcr.io/v2/homebrew/core/isl/manifests/0.26
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/ec0bbec77171645273dd59a9ccfdf290deccd999b3ca2082e0478d282b777ec0--isl-0.26.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/isl/blobs/sha256:91965ce2f54c7d1b16747ed05de989a3a122f5dbee67546bbf9bf065873b13c5
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/dd607d231aebbac4fa2e71906b408168440023c28b1ae3e536ac3f43b92154c2--isl--0.26.big_sur.bottle.tar.gz
==> Fetching mpfr
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/0912ebd653b1d6eb8fbf24dbf07e7bdab70ed6ff/Formula/m/mpfr.rb
################################################################################################################################################################## 100.0%
==> Downloading https://ftp.gnu.org/gnu/mpfr/mpfr-4.2.1.tar.xz
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/cf24a6e49aed02050761c208ba117f3cec9dbb744e6b31a6957561ec3ef84210--mpfr-4.2.1.tar.xz
==> Fetching libmpc
==> Downloading https://ghcr.io/v2/homebrew/core/libmpc/manifests/1.3.1
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/fdfa98e0f8bb3ce075cb32776ac2345aa2f89252706c162aecfc841085fa76be--libmpc-1.3.1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/libmpc/blobs/sha256:47b50c3df6a35ea3c876397eac4a7dc157b5f4109247671a16599a9a41b9c035
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/16f5080ab503971c18d01e013b6dcbf41068513dbc2a42d54e6d39f5139189a1--libmpc--1.3.1.big_sur.bottle.tar.gz
==> Fetching lz4
==> Downloading https://ghcr.io/v2/homebrew/core/lz4/manifests/1.9.4
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/379e59b981667f9585b33a2ff318769d8edca3ce6fd2e9a67ed291ae3e0cc872--lz4-1.9.4.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/lz4/blobs/sha256:aafb93487e108d302d060265898e4eaa82f5c806ff36dec50871db1c33fdc04d
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/8aa8e6aa5d3e84034501e402ebc2941cf2a77659908b9ba568871b494dd498d6--lz4--1.9.4.big_sur.bottle.tar.gz
==> Fetching cmake
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/0912ebd653b1d6eb8fbf24dbf07e7bdab70ed6ff/Formula/c/cmake.rb
################################################################################################################################################################## 100.0%
==> Downloading https://github.com/Kitware/CMake/releases/download/v3.29.3/cmake-3.29.3.tar.gz
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/cf3d7313c9722963e717c75c901b72588244fdf8f0e49b0b33b005e31c474944--cmake-3.29.3.tar.gz
==> Fetching zstd
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/0912ebd653b1d6eb8fbf24dbf07e7bdab70ed6ff/Formula/z/zstd.rb
################################################################################################################################################################## 100.0%
==> Downloading https://github.com/facebook/zstd/archive/refs/tags/v1.5.6.tar.gz
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/a5a961e2bd6292dafbd3b24b45f68dcfafe109b6fa81d7cdc52e6ab2df01363a--zstd-1.5.6.tar.gz
==> Fetching texinfo
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/0912ebd653b1d6eb8fbf24dbf07e7bdab70ed6ff/Formula/t/texinfo.rb
################################################################################################################################################################## 100.0%
==> Downloading https://ftp.gnu.org/gnu/texinfo/texinfo-7.1.tar.xz
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/e2cac89734353fb4827c98b77de17986414eb1ddb952cb4cf6464fb6fd24fc38--texinfo-7.1.tar.xz
==> Fetching mingw-w64
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/0912ebd653b1d6eb8fbf24dbf07e7bdab70ed6ff/Formula/m/mingw-w64.rb
################################################################################################################################################################## 100.0%
==> Downloading https://ftp.gnu.org/gnu/binutils/binutils-2.42.tar.xz
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/9a6410db5e2ec26298f4fa06e284ff22893706c8964572e146c313c066624344--binutils-2.42.tar.xz
==> Downloading https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=9970b576b7e4ae337af1268395ff221348c4b34a
Already downloaded: /Users/vagrant/Library/Caches/Homebrew/downloads/6c68aa4aef135895a658b1ab6d97313af8eadb8add341ff7a61b747317d00608--gcc.git-9970b576b7e4ae337af1268395ff221348c4b34a.patch
Error: mingw-w64: SHA256 mismatch
Expected: 968bfcb58f75d889470f2f815787f6aa254fb43f1e5516e04f577dad22259905
  Actual: 7f79e6bf59d58562990baaed5a69f83df57e6d6255a3ddf5c779320e58dd5f30
    File: /Users/vagrant/Library/Caches/Homebrew/downloads/6c68aa4aef135895a658b1ab6d97313af8eadb8add341ff7a61b747317d00608--gcc.git-9970b576b7e4ae337af1268395ff221348c4b34a.patch
To retry an incomplete download, remove the file above.
vagrant@osx-10_7 ~ %
```

### What did you expect to happen?

Brew to install successfully

### Step-by-step reproduction instructions (by running `brew` commands)

Command:

```
brew install mingw-w64
```

**Comment by mrpippy** [comment]
Argh I see this mismatch too, unfortunately I don't have previous copies of the patches to see what changed. There's a git version number on the bottom of the patch, I don't think that changed but maybe it did.
#171243 updates to GCC 14.1 and removes the need for these patches, I don't know if it's worth fixing these.

**Comment by ZhongRuoyu** [comment]
Should be fixed by #171243.
